uaclient/entitlements/repo.py

```diff
@@ -12,6 +12,7 @@ class RepoEntitlement(base.UAEntitlement):
 
     repo_list_file_tmpl = '/etc/apt/sources.list.d/ubuntu-{name}-{series}.list'
     repo_pref_file_tmpl = '/etc/apt/preferences.d/ubuntu-{name}-{series}'
+    origin = 'UNSET'   # The repo Origin value for setting pinning
```

Comment:
`None` would be a more normal value to indicate that something is unset; why the string?

Reply:
It's a good point, None would still be cast to 'None' in a print string. I just wanted something that would be a bit more visible/understandable given that RepoEntitlement.origin is written to user-facing files like /etc/apt/preferences.d/<some_pin_file>.

Reply:
That said; 'None' is still understandable from a config file, so I'm ok changing this to None. Thoughts

Reply:
If we write either `o=UNSET` or `o=None` to a config file, I would expect that config file to be useless. Neither of those origins are ever going to exist, so the pinning will never apply. Am I correct?

If so, I think we should ensure that we don't write out a config file if origin isn't something meaningful, and it's easier/clearer to check for None in the code that writes the file than it is 'UNSET'.

Reply:
+1 pushed changes to avoid emitting an o=None value. All entitlements have this origin hardcoded so we should be able to get here but unit tests are added for this logic.